File: star_pattern/star_pattern.py
iterations = 0
i = 0
size = 15
while iterations < size:
    iterations += 1
    if iterations <= size // 2:
        i += 1
        print(" " * (size - i), end="")
        print("* " * i)
    else:
        i -= 1
        print(" " * (size - i), end="")
        print("* " * i)

File: star_pattern/StarPattern.java
public class StarPattern {
	public static void main(String[] args) {
		int iterations = 0;
		int i = 0;
		int size = 15;

		while (iterations < size) {
			iterations++;
			if (iterations <= (int) (size / 2)) {
				i++;
				System.out.print(" ".repeat(size - i));
				System.out.println("* ".repeat(i));
			} else {
				i--;
				System.out.print(" ".repeat(size - i));
				System.out.println("* ".repeat(i));
			}
		}
	}
}